modules/dnn/perf/perf_net.cpp

```diff
@@ -29,7 +29,7 @@ class DNNTestNetwork : public ::perf::TestBaseWithParam< tuple<Backend, Target>
     }
 
     void processNet(std::string weights, std::string proto, std::string halide_scheduler,
-                    const Mat& input, const std::string& outputLayer = "")
+                    const Mat& input, const std::string& outputLayer = "", const std::string& data_dim="hwc")
```

Comment:
I'd recommend change method behavior instead of adding a new argument:
```cpp
void processNet(std::string weights, std::string proto, std::string halide_scheduler,
                    const Mat& input, const std::string& outputLayer = "")
//...
net.setInput(input);
```
and add an override:
```cpp
void processNet(std::string weights, std::string proto, std::string halide_scheduler,
                     Size inpSize, const std::string& outputLayer = "")
Mat inp(inpSize, CV_32FC3);
processNet(weights, halide_scheduler, inp, outputLayer);
```

Reply:
Thanks! I will modify it.